src/board/BoardLogic.java

```diff
@@ -289,23 +297,66 @@ protected boolean removeApple(int x, int y) {
      * place apples on the board until MAX_APPLES_ON_BOARD value is reached
      */
     protected void setApples() {
+        // get all valid fields
+        List<Field> validFields = new ArrayList<>(getValidAppleFields());
 
         // add apples until value of MAX_APPLES_ON_BOARD is reached
-        while (apples.size() < MAX_APPLES_ON_BOARD) {
-
-            // get random field with field status empty
-            Field appleField;
-            do {
-                appleField = getRandomField();
-
-            } while (!(fields[appleField.getPosX()][appleField.getPosY()].getState() == FieldState.Empty));
+        while (apples.size() < MAX_APPLES_ON_BOARD && validFields.size() > 0) {
+            // get random valid field
+            Field appleField = validFields.get((int) (Math.random() * validFields.size()));
```

Comment:
This may produce an index out of bounds exception

Reply:
How? Math.random returns [0, 1) and int cast practically rounds down.

Reply:
please use the Random.nextint() to avoid rounding errors introduced by float multiplication

Reply:
If we use Random we should also use it in the generate random field function and also maintain a somewhat global Random instance, but I honestly don't really see the point where this would ever be an issue

Reply:
It is bad practice: https://stackoverflow.com/questions/738629/math-random-versus-random-nextintint
and can lead to unexpected result and weird behavior and a non uniform distrobution.
I have to admit, no critical errors will be produced, since our value will in this case always be rounded down. But e.g.
int((1- 2**-53) * 928374987928374096) = 928374987928374016
In that case some of our values can't even be reached

Reply:
I will a Random instance ^^